### PR TITLE
feat(runtime): allow to automatically set GOMAXPROCS according to linux container quota

### PR DESCRIPTION
**Pull request by drwatsno**
## Related issue
#2033 

## Proposed changes

ADD cgroups.v1.auto_max_procs_enabled option. Defaults to false
If true, sets GOMAXPROCS automatically according to cpu quota

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security. vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [x] I have added or changed [the documentation](docs/docs).

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -84,6 +84,7 @@ func initConfig() {
 	}
 
 	viper.SetDefault(configuration.ViperKeyLogLevel, "info")
+	viper.SetDefault(configuration.ViperKeyCGroupsV1AutoMaxProcsEnabled, false)
```
Comment (aeneasr) [inline cmd/root.go:87]: No need to set this when the default is configured in the config schema :)

File: cmd/server/handler.go [hunk 2]
```diff
@@ -181,6 +182,14 @@ func RunServeAll(version, build, date string) func(cmd *cobra.Command, args []st
 func setup(d driver.Driver, cmd *cobra.Command) (admin *x.RouterAdmin, public *x.RouterPublic, adminmw, publicmw *negroni.Negroni) {
 	fmt.Println(banner(d.Registry().BuildVersion()))
 
+	if d.Configuration().CGroupsV1AutoMaxProcsEnabled() {
+		_, err := maxprocs.Set(maxprocs.Logger(d.Registry().Logger().Infof))
+
+		if err != nil {
+			panic(err)
```
Comment (aeneasr) [inline cmd/server/handler.go:189]: Please use a fatal instead, see:

```
d.Registry().Logger().WithError(err).Fatal("...")
```

**Review by aeneasr — COMMENTED** [review]
Awesome!

File: cmd/root.go (hunk 1 above)
Comment (drwatsno) [reply cmd/root.go]: removed this line

File: cmd/server/handler.go (hunk 2 above)
Comment (drwatsno) [reply cmd/server/handler.go]: replaced with `d.Registry().Logger().WithError(err).Fatal("Couldn't set GOMAXPROCS")`

**Comment by drwatsno** [comment]
@aeneasr it looks like tests are not needed

**Review by aeneasr — APPROVED** [review]

**Comment by aeneasr** [comment]
Awesome, thank you! :)

**Pull request merged**